src/dotnet-roslyn-tools/PRTagger/PRTagger.cs

```diff
@@ -188,40 +311,125 @@ private static async Task<GitRepository> GetVSRepositoryAsync(GitHttpClient gitC
         return null;
     }
 
-    private async static Task TryCreateIssueAsync(
+    private static async Task<TagResult> TryCreateIssueAsync(
+        HttpClient client,
+        string title,
         string gitHubRepoName,
-        string vsBuildNumber,
         string issueBody,
-        string gitHubToken,
         ILogger logger)
     {
-        var client = new HttpClient
-        {
-            BaseAddress = new("https://api.github.com/")
-        };
-
-        var authArray = Encoding.ASCII.GetBytes($"{gitHubToken}");
-        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
-            "Basic",
-            Convert.ToBase64String(authArray));
-        client.DefaultRequestHeaders.Add(
-            "user-agent",
-            "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2;)");
 
         // https://docs.github.com/en/rest/issues/issues#create-an-issue
         var response = await client.PostAsyncAsJson($"repos/dotnet/{gitHubRepoName}/issues", JsonConvert.SerializeObject(
             new
             {
-                title = $"[Automated] PRs inserted in VS build {vsBuildNumber}",
+                title = title,
                 body = issueBody,
-                labels = new string[] { "vs-insertion" }
+                labels = new string[] { InsertionLabel }
             }));
 
         if (!response.IsSuccessStatusCode)
         {
             logger.LogError($"Issue creation failed with status code: {response.StatusCode}");
+            return TagResult.Failed;
         }
 
         logger.LogInformation("Successfully created issue.");
+        return TagResult.Succeed;
+    }
+
+    /// <summary>
+    /// Check if the issue with <param name="title"/> exists in repo.
+    /// </summary>
+    private static async Task<bool> HasIssueAlreadyCreatedAsync(
+        HttpClient client,
+        string repoName,
+        string title,
+        ILogger logger)
+    {
+        var response = await client.GetAsync($"search/issues?q={title}+repo:dotnet/{repoName}+is:issue+label:{InsertionLabel}").ConfigureAwait(false);
```

Comment:
Can this just call into `SearchIssuesOnGitHubAsync` so that there is only one bit of code responsible for searching issues?